### PR TITLE
Profile page

### PR DESCRIPTION
**Pull request by foolhardy21**
## Profile Page.

This PR has APIs and UI components related to the profile page.

- Files to review:
   - [ ] src/pages/Profile.jsx
   - [ ] src/contexts/profile.context.js
   - [ ] src/reducers/profile.reducer.js
   - [ ] src/components/Profile/ProfileBio.jsx
   - [ ] src/components/Profile/PostEdit.jsx
   - [ ] src/components/Profile/ProfileEdit.jsx

- Use this link for preview - https://deploy-preview-9--remarkable-flan-74a807.netlify.app/login

v-2
v-1

Thanks

### CODE REVIEW

**Review by abhisheksahani-nova — APPROVED** [review]
Looks good and clean , nothing to suggest good to merge

File: src/components/Profile/ProfileBio.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                <a href={bio?.portfolio} target='_blank' className='txt-md txt-secondary flx flx-min-center'>
+                    <span className="material-icons txt-off-secondary mg-right-xs">
+                        link
+                    </span>
+                    {bio.portfolio ?? 'your portfolio'}
+                </a>
+            </div>
+            <div className='flx flx-maj-stretch flx-min-center mg-top-md'>
+                <div className='flx flx-min-center'>
+                    <p className='txt-md txt-secondary txt-cap mg-right-xs'>{`followers ${bio?.followers?.length}`}</p>
```
Comment (nayyyhaa) [inline src/components/Profile/ProfileBio.jsx:89]: appreciation: good that you are using optional chaining here.

File: src/components/Profile/PostEdit.jsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const { editPost, postsDispatch } = usePosts()
+
+    const handlePostEdit = async () => {
+        const response = await editPost(id, post)
+        if (response.status === 201) {
+            const editedPost = response.data.posts.find(newpost => newpost._id === post._id)
+            postsDispatch({ type: ACTION_EDIT_POST, payload: editedPost })
+        }
+        setPost({})
+        setModal(m => ({ ...m, type: '', id: '' }))
```
Comment (nayyyhaa) [inline src/components/Profile/PostEdit.jsx:19]: suggestion: try to be more descriptive with var naming.

**Review by nayyyhaa — APPROVED** [review]
Only one minor suggestion, rest your code looks clean and well structured! You can merge the changes 👍

File: src/components/Profile/PostEdit.jsx (hunk 2 above)
Comment (foolhardy21) [reply src/components/Profile/PostEdit.jsx:19]: Sure, I'll make the changes.

**Pull request merged**